core/src/test/java/org/pentaho/di/core/row/value/ValueMetaBaseTest.java

```diff
@@ -1111,4 +1117,368 @@ public void testHashCodes() throws Exception {
     valueMetaString.type = ValueMetaInterface.TYPE_NONE;
     assertEquals( valueMetaString.hashCode( "any" ),  0 );
   }
+
+  @Test
+  public void testGetMetadataPreview() throws SQLException, KettleDatabaseException {
+    ValueMetaBase valueMetaBase = new ValueMetaBase( );
+
+    DatabaseMeta dbMeta = spy( new DatabaseMeta() );
+
+    // SQL CHAR is a Pentaho String
+    ResultSet resultSet = mock( ResultSet.class );
+    Mockito.when( resultSet.getInt( "DATA_TYPE" ) ).thenReturn( Types.CHAR );
```

Comment:
Please use the syntax `doReturn( Types.CHAR ).when( resultSet ).getInt( "DATA_TYPE" )` instead.
This syntax has given use some problems in the past with tests failing for no apparent reason.
Also use an `import static` instead of `Mockito.method` this is customary for unit tests.